mod/workshop/locallib.php

```diff
@@ -3447,122 +3359,161 @@ protected function aggregate_grading_grades_process(array $assessments, $timegra
     }
 
     /**
-     * Returns SQL to fetch all enrolled users with the given capability in the current workshop
-     *
-     * The returned array consists of string $sql and the $params array. Note that the $sql can be
-     * empty if a grouping is selected and it has no groups.
+     * Gets users with any of the specified capabilities on the workshop activity.
      *
      * The list is automatically restricted according to any availability restrictions
      * that apply to user lists (e.g. group, grouping restrictions).
      *
-     * @param string $capability the name of the capability
-     * @param bool $musthavesubmission ff true, return only users who have already submitted
+     * @param array $capabilities array of capability names (If the user has ANY of them, it returns true for that user)
+     * @param bool $musthavesubmission if true, return only users who have already submitted
      * @param int $groupid 0 means ignore groups, any other value limits the result by group id
-     * @return array of (string)sql, (array)params
+     * @param int $limitfrom return a subset of records, starting at this point (optional, required if $limitnum is set)
+     * @param int $limitnum return a subset containing this number of records (optional, required if $limitfrom is set)
+     * @return array of users
      */
-    protected function get_users_with_capability_sql($capability, $musthavesubmission, $groupid) {
```

Comment:
Need to add param and return types.

Reply:
This is removed, but I've added them in the replacement function I added instead.